packages/test/test-utils/package.json

```diff
@@ -133,6 +132,7 @@
 		"@fluidframework/routerlicious-driver": "workspace:~",
 		"@fluidframework/runtime-definitions": "workspace:~",
 		"@fluidframework/runtime-utils": "workspace:~",
+		"@fluidframework/synthesize": "workspace:~",
```

Comment:
We probably shouldn't have this dependency in test-utils either

Reply:
Probably not, but needed for now (unless someone can advise me on how to remove the lingering use). Note that this already existed as a transitive dep, so this isn't a regression.

Reply:
Just kidding. This is no longer needed after some refactoring I already did. Removed!

Reply:
I had to restore this to fix the typing of `ContainerRuntimeFactoryWithDefaultDataStoreProps.dependencyContainer`.

Reply:
No code in the repo currently uses it, but the comment in `testContainerRuntimeFactoryWithDefaultDataStore.ts` seems to imply there are back-compat requirements involved here. It isn't clear to me if those are real (given that the API is internal) or if they are still relevant. I don't know if/how this code is involved in any cross-version compat testing.

Reply:
@kian-thompson your name came up in a git=blame in this space. Any chance you can offer some context here?

Reply:
A question relating to this helper util just came up in my team. Basically, the E2E infra only uses the current version's `ContainerRuntimeFactoryWithDefaultDataStore` type, while at run time we may dynamically load an older version and still try to apply the new type's definition to the old version.

Either the `createContainerRuntimeFactoryWithDefaultDataStore(...)` util needs to stay, or the bug in the test infra needs to be fixed.

Reply:
Thanks for the context, Kian!

Reply:
So we can't remove the property. But since we don't have any usages in the repo, and don't want any new ones, I have:
1. Updated the typing to use `never`
2. Marked the member as deprecated with a note about existing only for back-compat
3. Removed the dependency on `synthesize`